### PR TITLE
Add simple Python script to convert NCEPLIBS lua modulefiles into tcl modulefiles

### PR DESCRIPTION
**Pull request by climbfuji**
As the title says. Not a generic converter, but no more or no less than what is required the NCEPLIBS lua modulefiles for dinosaur systems.

### CODE REVIEW

**Comment by aerorahul** [comment]
@climbfuji 
Thanks for this. 
Can you make this part of the cmake?

In particular, you could add a `-DTCL=ON` option (default: OFF) in CMakeLists.txt, `make tcl` a target (if `TCL=ON`), and update `deploy_modulefiles.sh.in` to deploy tcl or lmod modulefiles.

I am willing to help.

**Comment by climbfuji** [comment]
The way this works is that it needs to parse the _final_ lua modulefiles (i.a. after deploying them) and then create tcl  modulefiles next to the lua modulefiles. Systems that don't speak lua do not see the lua modulefiles, hence no conflict.
```
./lua2tcl.py /directory/to/which/the/lua/modulefiles/where/deployed
```

**Comment by climbfuji** [comment]
I merged @aerorahul's update that provides an option to cmake for calling the Python script post-install.

**Review by aerorahul — APPROVED** [review]
looks good. 
Thanks for adding the option.

File: CMakeLists.txt [hunk 1]
```diff
@@ -189,4 +190,11 @@ if(DEPLOY)
   add_custom_target(deploy
     COMMAND ${CMAKE_CURRENT_BINARY_DIR}/deploy_modulefiles.sh)
   add_dependencies(deploy prep2deploy)
+  if(TCLMOD)
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/lua2tcl.py
+                   ${CMAKE_CURRENT_BINARY_DIR}/lua2tcl.py COPYONLY)
+    add_custom_target(tcl
+      COMMAND ${CMAKE_CURRENT_BINARY_DIR}/lua2tcl.py ${CMAKE_INSTALL_PREFIX}/modules)
```
Comment (DusanJovic-NOAA) [inline CMakeLists.txt]: Do you need to copy this script to binary dir, can't you run it from  `CMAKE_CURRENT_SOURCE_DIR` directly?

Comment (climbfuji) [reply CMakeLists.txt]: You can run it from everywhere, as long as you provide the full path (relative or absolute) to the installed modules.

Comment (aerorahul) [reply CMakeLists.txt]: One can run from the source dir.  I like to copy from source to binary before I make changes (if needed). 
 Ideally, one would place it in `bin`, but that would be too radical for some. This is how I would do it. Feel free to make the change.

Comment (climbfuji) [reply CMakeLists.txt]: I leave it up to you two - if I have to choose, I'll prefer to run from the source dir.

Comment (climbfuji) [reply CMakeLists.txt]: How about now?

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by mark-a-potts** [comment]
These worked perfectly for me on my test system.

**Pull request merged**